pkg/stripe/client.go

```diff
@@ -90,7 +90,12 @@ func (c *Client) PerformRequest(ctx context.Context, method, path string, params
 	// RequestID of the API Request
 	requestID := resp.Header.Get("Request-Id")
 	livemode := strings.Contains(c.APIKey, "live")
-	go sendTelemetryEvent(ctx, requestID, livemode)
+
+	// Weird edge case: ctx can be nil if the user runs a command like `stripe products list` but isn't logged in
```

Comment:
I'd say this is alright as a hotfix, but context should really almost never be passed around as `nil`. The real bug is likely somewhere else, where the correct context is either not being propagated, or, if it's not available, it's not being set to [context.Background](https://pkg.go.dev/context#Background) or [context.TODO](https://pkg.go.dev/context#TODO).

If we're not sure where that is (it should be discoverable from the panic stackstace), this is good for now and I'd be happy to pair on a followup task to trace down the source of the nil!

Reply:
Yup, you're right and I am looking into the root cause.

It's coming from here: https://github.com/stripe/stripe-cli/blob/0c97aabe2b0602a4b7f95a50df5018dd0279368c/pkg/cmd/root.go#L97

Right now I'm trying to understand how we are supposed to correctly invoke this command with context.
```
loginCommand.ExecuteContext(updatedCtx);
```
does not do what I thought it would

Reply:
One thing we can do is to replace that line with https://github.com/stripe/stripe-cli/blob/0c97aabe2b0602a4b7f95a50df5018dd0279368c/pkg/cmd/login.go#L43 directly rather than calling `Run` or `Execute`. WDYT?

Reply:
Okay, the reason `loginCommand.ExecuteContext(updatedCtx);` doesn't work is because the cobra command is still for trigger and all this does is try to run the trigger command again. I think your solution makes more sense than us trying to mimick another cobra invocation, Vincent!